cloudmetrics/metrics/fourier.py

```diff
@@ -0,0 +1,573 @@
+    assued to be square, computes the spectral power of this field and
+    decomposes this into radial and azimuthal power spectral densities.
+
+    Parameters
+    ----------
+    cloud_scalar : numpy array of shape (npx,npx) - npx is number of pixels
+            Cloud mask field.
+    dx : int, optional
+        Horizontal (uniform) grid spacing, for computing wavenumbers. The
+        default is 1.
+    periodic_domain : Bool, optional
+        Whether the domain is periodic. If False, choose an appropriate window
+        to apply before performing the FFT. The default is False.
+    apply_detrending : Bool, optional
+        Whether to remove a blinear fit of the input scalar field. The default
+        is False.
+    window : String, optional
+        Which windowing function to apply. Set to a value in
+        ['Hann', 'Welch', 'Planck']. The default is None, corresponding to no
+        applied windowing.
+
+    Returns
+    -------
+    k1d : 1D numpy array
+        Radial 1D wavenumbers of the FFT, whose wavelength is 2*pi/k1d
+    psd_1d_rad: 1D numpy array
+        Radial 1D power spectral density of the cloud_scalar field
+    psd_1d_azi: 1D numpy array
+        Azimuthal 1D power spectral density of the cloud_scalar field
+    """
+
+    # TODO: This explicitly assumes square domains
+
+    # General observations
+    # Windowing   : Capturing more information is beneficial
+    #               (None>Planck>Welch>Hann window)
+    # Detrending  : Mostly imposes unrealistic gradients
+    # Using image : Less effective at reproducing trends in 2D org plane
+    # Binning     : Emphasises lower k and ignores higher k
+    # Assumptions : 2D Fourier spectrum is isotropic
+    #               Impact of small-scale errors is small
+
+    # Detrending
+    if apply_detrending:
+        [X, Y] = np.meshgrid(
+            np.arange(cloud_scalar.shape[0]),
+            np.arange(cloud_scalar.shape[1]),
+            indexing="ij",
+        )
+        cloud_scalar, bDt = _detrend(cloud_scalar, [X, Y])
+
+    # Windowing
+    if periodic_domain:
+        if window == "Planck":
+            cloud_scalar = _planck_rad(cloud_scalar)  # Planck-taper window
+        elif window == "Welch":
+            cloud_scalar = _welch_rad(cloud_scalar)  # Welch window
+        elif window == "Hann":
+            cloud_scalar = _hann_rad(cloud_scalar)  # Hann window
+
+    # FFT
+    F = fftpack.fft2(cloud_scalar)  # 2D FFT (no prefactor)
+    F = fftpack.fftshift(F)  # Shift so k0 is centred
+    psd_2d = np.abs(F) ** 2 / np.prod(cloud_scalar.shape)  # Energy-preserving 2D PSD
+    psd_1d_rad = _get_psd_1d_radial(psd_2d, dx)  # Azimuthal integral-> 1D radial PSD
+    psd_1d_azi = _get_psd_1d_azimuthal(psd_2d)  # Radial integral -> Sector 1D PSD
+
+    # Wavenumbers
+    N = np.min(cloud_scalar.shape)
+    L = dx * N
+    k1d = 2 * np.pi / L * np.arange(1, N // 2 + 1)
+
+    return k1d, psd_1d_rad, psd_1d_azi
```

Comment:
I was wondering whether it might be worthwhile returning a `xr.DataSet` here instead? That way we can use the `k1d` variable to define the coordinate values. That would make it easier someone to reason about what the return values are. What do you think about that?

Reply:
It's a good idea. If we eventually want to hide all this stuff anyway, it might not matter much; if we don't, then I like the idea of clarifying what is a coordinate. I don't know if people will find ``xr.DataSet``s more daunting to deal with than just the arrays, though? For now, I've chosen to make the docstring of ``compute_spectra`` be really clear about ``k1d`` being ``psd_1d_rad``'s coordinate, hope that's okay.